src/core/rules.rkt

```diff
@@ -36,16 +37,23 @@
 (define (make-rule-context input output)
   (map (curryr cons 'real) (set-union (free-variables input) (free-variables output))))
 
+(define (add-unsound expr)
+  (match expr
+    [(list op args ...) (cons (sym-append "unsound-" op) (map add-unsound args))]
```

Comment:
Ah, interesting, you're doing this automatically, not by hand. Possibly we could do better by hand-selecting which operations are actually unsafe, but maybe this way is easier.